open_academy/models/course.py

```diff
@@ -10,3 +10,21 @@ class Course(models.Model):
     description = fields.Text()
     responsible_user_id = fields.Many2one('res.users')
     session_ids = fields.One2many('session', 'course_id')
+    _sql_constraints = [
+        ('unique_course_name', 'UNIQUE (name)', 'This course already exists'),
+        (
+            'check_title_description',
+            'CHECK (title NOT LIKE description)',
+            'The title must be different than the description'
+        )
+    ]
+
+    def copy(self, default=None):
+        default = default or {}
+        copied_count = self.search_count([('name', '=like', 'Copy of %s' % self.name)])
+        if not copied_count:
```

Comment:
You can skip the evaluation of the else with something like 

```suggestion python 
        new_name =  'Copy of %s' % self.name
        new_name += '(%s)' % copied_count if copied_count > 0 else ''
```
Please try and test this

Reply:
Thanks for the suggestion @andreaarce I really like this approach